lib/cf-socket.c

```diff
@@ -1153,8 +1142,8 @@ static ssize_t cf_socket_recv(struct Curl_cfilter *cf, struct Curl_easy *data,
 
   DEBUGASSERT(data->conn == cf->conn);
   nread = Curl_recv_plain(data, cf->sockindex, buf, len, err);
-  CF_DEBUGF(infof(data, CFMSG(cf, "recv(len=%zu) -> %d, err=%d"),
-            len, (int)nread, *err));
+  DEBUGF(LOG_CF(data, cf, "recv(len=%zu) -> %d, err=%d", len, (int)nread,
+                *err));
```

Comment:
I appreciate why this may be necessary temporarily, and useful, but even for me this is sometimes too verbose. It looks like this PR is going to change CF logging so that it cannot be disabled on DEBUGBUILDs any longer.

Reply:
Theses statements will not produce output, unless you enable the logging for a certain cfilter type.

```
CURL_DEBUG=tcp curl -v url
```

will enable the logging of the 'TCP' socket cfilter. 

The idea here is to have logging available, should you wish, without recompiling the debug-enabled curl.

Reply:
Neat. We will need documentation explaining how it works.